controllers/tutorial.js

```diff
@@ -286,13 +286,21 @@ const forgotPassword = async (req, res) => {
 const resetPassword = async (req, res) => {
   try {
     const resultValidated = await resetSchema.validateAsync(req.body);
-    const resetdb = await user.findOne({ email: resultValidated.email });
+    const resetdb = await reset.findOne({ email: resultValidated.email, otp: resultValidated.otp });
     if (!resetdb) {
       return res.status(400).send('Please generate a otp or check the mail');
     }
+    // validate old password
+    // const isValidPassword = await bcrypt.compare(user.oldPassword, resultValidated.newPassword);
+
     // encrypt the password
     const salt = await bcrypt.genSalt(10);
     const hashedPassword = await bcrypt.hash(resultValidated.newpassword, salt);
+
+    const passwordExist = await reset.findOne({ password: hashedPassword });
```

Comment:
reset contains fields like email , OTP, and the new password , if I'm not wrong 
old password would be stored in user schema, please change this code 
also, there would be a problem while comparing normal string to hashed string 
kindly use 
`bcryptjs.compare(string,hash)`
by this, you can perform this operation